src/Confluent.Kafka/Producer.cs

```diff
@@ -671,7 +673,41 @@ internal Producer(ProducerBuilder<TKey, TValue> builder)
             }
 
             this.ownedKafkaHandle = SafeKafkaHandle.Create(RdKafkaType.Producer, configPtr, this);
-            configHandle.SetHandleAsInvalid(); // config object is no longer usable.
+            configHandle.SetHandleAsInvalid();  // ownership was transferred.
+                                                // configPtr remains valid over the lifetime of this method.
+
+            Func<PartitionerDelegate, IntPtr> createTopicConfigWithPartitioner = (partitioner) => {
+                var topicConfigPtr = SafeTopicConfigHandle.DuplicateDefaultTopicConfig(configPtr);
```

Comment:
Shouldn't this be `DuplicateDefaultTopicConfig(this.ownedKafkaHandle)`?

Reply:
how could this have possibly been working is the question on my mind.